programs/mpl-core/src/plugins/oracle.rs

```diff
@@ -0,0 +1,13 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::ExtraAccount;
+
+/// Oracle plugin that allows getting a `ValidationResult` for a lifecycle event from an arbitrary
+/// account either specified by or derived from the `Pubkey` attached to the `ExternalPluginKey`.
+/// This hook is used for any lifecycle events that were selected in the `ExternalPluginRecord` for
+/// the plugin.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub struct Oracle {
+    /// Optional PDA (derived from Pubkey attached to `ExternalPluginKey`).
+    pub pda: Option<ExtraAccount>,
```

Comment:
It can be any account, which is specified in the key.

So the address specified in the key is the base address, and either
1. that base address is used directly as the address to read from, or
2. if the user puts in the PDA option, then that means the base address is a program ID and use the ExtraAccount to derive a PDA from it.

Reply:
I see ok cool